### DISCUSSION TITLE
Can I get CsvHelper to determine the difference between two CSV based on column/headers alone?

### DISCUSSION THREAD

**Discussion by CaiusJard** [JoshClose/CsvHelper#1816] [Q&A]
Suppose I have two CSV that arrive with a guid filename, randomly. The only way I can tell it it's an "A" file or a "B" file is by inspecting the columns - an A file always has 3 columns (named X, Y, Z) and a B file always has 2 columns (named L, M)

I'm using e.g. for just loading A type files:

                using (var reader = new CsvReader(new StreamReader(streamFromSocket, System.Text.Encoding.UTF8), conf))
                {

                    AModel reusedViewModel = new AModel();
                    await foreach (var ignore in reader.EnumerateRecordsAsync<AModel>(reusedViewModel))

I now want to extend this and inspect the header of the file and open an `EnumerateRecordsAsync<BModel>`  if it's a B file. Any pointers on how I do that in this particular scenario of enumerating records from the socket?

**Comment by JoshClose** [comment]
```cs
using (var csv = new CsvReader(reader, config))
{
    csv.Read();
    csv.ReadHeader();
    var numberOfColums = csv.Parser.Count;
    while (csv.Read())
    {
        if (numberOfColums == 2) 
        {
            csv.GetRecord<BModel>();
        }
        else if (numberOfColums == 3)
        {
            csv.GetRecord<AModel>();
        }
    }
}
```
